autoload/quickfixsigns/vcsdiff.vim

```diff
@@ -573,6 +555,33 @@ function! s:GetParam(name, type, default) abort "{{{3
 endf
 
 
+function! s:Diff(filename, vcs_type) "{{{3
+  let cmdt    = s:Config(a:vcs_type).cmd
+  let rev_arg = s:GetParam('quickfixsigns#vcsdiff#rev_arg', a:vcs_type, s:Config(a:vcs_type).rev_arg)
+  let extra   = s:GetParam('quickfixsigns#vcsdiff#extra_args', a:vcs_type, '')
+  let dir     = fnamemodify(a:filename, ':h')
+  let file    = fnamemodify(a:filename, ':t')
+
+  " Create command string
+  let cmds = printf("%s %s", g:quickfixsigns#vcsdiff#cd, shellescape(dir))
+  if exists('g:quickfixsigns#vcsdiff#revision') && !empty('g:quickfixsigns#vcsdiff#revision')
```

Comment:
g:quickfixsigns#vcsdiff#revision should be queried as user-configurable parameter.

I'd rather set revision to the formatted revision argument if a revision is specified or an empty string if no rev is specified. This makes the code duplication in the else branch obsolete.